README.md

```diff
@@ -43,11 +44,15 @@ contract MyContract is BitmaskRBACPermissionable{
 To create a new user, just specify the name (required) and bitmask of suitable roles:
 
 ```
-await rbac.newUser(admin, 'Contract Creator', ADMIN | PUBLISH | VALIDATE)
+await rbac.newUser(user_address, 'Contract Creator', ADMIN | PUBLISH | VALIDATE)
 ```
 
+## Usage (Javascript)
+
+Once you have imported your user roles into javascript (e.g. using drizzle), it is super easy to define appropriate constants (e.g. `ADMIN = 1; PUBLISH = 2; VALIDATE = 4`) and then check whether a user bitmask allows a given role or roles. E.g. `(user.role & PUBLISH) !== 0` to check if a user can publish, or even `(user.role & (VALIDATE | ADMIN)) !== 0` to check if a user can Validate or Admin.
+
 ## TODO
 
-Scope what roles a user can add/remove to their role(s)
-Investigate gas use when there are many (> 100) roles, and consider replacing openzeppelin rbac dependency with logic that uses bitmasks natively
-Prevent duplicate roles
+- Ensure we cannot accidentally delete the last user with an `rbac_admin` role, or else the RBAC becomes fixed forever.
```

Comment:
this is complete, right?

Reply:
Yeah. See last two commits for comments on gas usage but if we're both happy with the user counts and admin preservation stuff then this should come out.